### DISCUSSION TITLE
3.0.0 - AS4CryptoProperties deprecation breaks ENV  property loading

### DISCUSSION THREAD

**Discussion by oaxfreeman** [phax/phase4#291] [Q&A] [answered]
Version 3.0.0 introduced new way for handling properties which breaks standard of `ENV_VAR`:`default_value`.
@phax can you provide some guidance on how to handle `application.properties` in the new release?
![image](https://github.com/user-attachments/assets/cd634ebd-7e29-4ca0-9b86-870fa8ec39e3)

**Comment by phax** [comment, marked as answer]
This is a common problem.
The way you are using it, follows the Spring Boot approach for variables: `${property:default-value}`.
However, as phase4 is independent of Spring Boot, it also uses its own configuration system as defined in https://github.com/phax/ph-commons/wiki/ph-config and does not allow the provision of default value.
By accident (or unfortunately), Spring Boot and phase4 both share the name `application.properties` for their default configuration files. However, the resolution of the properties is different in the Spring part and in the phase4 part, which regularly creates issues.

So I created https://github.com/phax/ph-commons/issues/44 to add this functionality

**Reply by phax** [reply]
The issue was resolved in ph-commons 11.1.11 and will be part of the phase4 3.0.1 release.
Please see https://github.com/phax/ph-commons/wiki/ph-config#variable-default-values for the detailed explanation
